### ISSUE TITLE
Screen Resolution keeps Resetting in Project 64

### ISSUE THREAD

**Issue by TradeDraft** [gonetz/GLideN64#1843]
When I change the Screen Resolution it goes back to Default when I exit game in go back in or play new game.

**Comment by gonetz** [comment]
If you have "Use per-game settings" option enabled and changes resolution or any other option when game running, changed settings stored per-game. That is, your new settings will work only for this game.
When you stop the game, settings restored to current profile. When you run new game, plugin loads settings for this new game.

So, you need to set your preferred options before you run any game. You may create several profiles with different preferences. Then, if you need to make adjustments in settings for some game, you run this game, open configuration dialog, change settings and save them.

**Comment by TradeDraft** [comment]
Thanks. That fixed that Problem.

Though I use different Video Plug-Ins so can be Forgotten to do it sometimes and This only happened with Latest Version.

Thanks for the Help Mate

**Comment by gonetz** [comment]
What is "Default profile"?

I may add a message box to ask user, where to save settings: current profile or custom settings for currently running game.  Of course, this question will be asked only if a game is running and "per game settings" option is enabled.

**Comment by TradeDraft** [comment]
That might be a Very Good Idea to ahve a Message Box where you want to save the Changes too.

For a Profile that effects all games or Custom Setting to only games you want to change setting for

**Comment by Aquatakat** [comment]
@gonetz The default profile would be the global save settings, what happens when you edit settings without having a game loaded. Maybe the language can be changed.

I'm just trying to avoid the distasteful feeling of going through the process of changing settings then pressing OK and then the dialogue box asking you more questions.

**Comment by gonetz** [comment]
@Aquatakat There is no default profile anymore. User may use arbitrary number of profiles. Config settings correspond to currently selected profile.
I agree that additional dialog box is not a best design solution. Additional control, as on your screen shot, can be better. Actually there are only two choices for user: "Save settings to Current Profile" or "Save as per-game settings". I added possibility to save as per-game settings only recently because it is inconvenient for user (and for me too) to edit custom ini in text editor. From the other side, it leads to confusions. That problem can be solved with additional dialog box or additional controls. Since there are only two options, it can be more illustrative to use radio boxes instead of combo box.

**Comment by gonetz** [comment]
@Aquatakat during work on #1879, I realized that it is not a good idea to give user option where to save settings. If per-games settings enabled, plugin loads settings from the current profile and then overrides them with per-game settings. Suppose, profile set resolution 640x480, but per-game settings use resolution 960x720. User run the game, it will start in 960x720. User opens config dialog during game play and changes resolution to 1024x768. What plugin should do if user will select "Save to my profile"? Set profile settings to 1024x768 but keep running in 960x720?
To avoid such collisions, I decided to allow save settings only for current game if per-game is enabled and game is running. I added a warning for user about it: 7d026f069
